python/semantic_kernel/connectors/memory/weaviate/weaviate_collection.py

```diff
@@ -148,47 +169,137 @@ async def _inner_upsert(
     ) -> Sequence[TKey]:
         assert all([isinstance(record, DataObject) for record in records])  # nosec
 
-        async with self.async_client:
-            try:
-                collection: CollectionAsync = self.async_client.collections.get(self.collection_name)
-                response = await collection.data.insert_many(records)
-            except Exception as ex:
-                raise MemoryConnectorException(f"Failed to upsert records: {ex}")
+        try:
+            collection: CollectionAsync = self.async_client.collections.get(self.collection_name)
```

Comment:
Without the context manage, don't we need to call `.connect()` manually: https://weaviate.io/developers/weaviate/client-libraries/python/async#context-manager?

Reply:
yes, we do in our aenter

Reply:
Same question, what if people don't use the context manager? That shouldn't stop people from being able to use the connector.

Reply:
this is the same for other collections, and it will throw, but I think we need to steer people to using the context manager, since it is much cleaner anyway, I will update docs accordingly

Reply:
The thing about context manager is that users will have to use the connector within a `with` block. Similar to the clients we are using, we need to do special handlings to use them properly. I am not sure if this is the direction we are going, but it seems counter intuitive to take away the ability to use our connectors outside a context manager, i.e. what if we need to pass the connector around in the kernel.

Reply:
I agree people should use context manage as much as possible but we should still allow them to use the connectors without a context manager.